sdks/java/core/src/test/java/org/apache/beam/sdk/runners/TransformTreeTest.java

```diff
@@ -159,7 +159,9 @@ public void visitTransform(TransformTreeNode node) {
         assertThat(transform, not(instanceOf(Sample.SampleAny.class)));
         assertThat(transform, not(instanceOf(Write.Bound.class)));
         if (transform instanceof Read.Bounded) {
-          assertTrue(visited.add(TransformsSeen.READ));
+          assertTrue(
+            node.getEnclosingNode().getTransform() instanceof TextIO.Read.Bound
+                == visited.add(TransformsSeen.READ));
```

Comment:
Create is used within the `Write` transform, and now that Create also uses `Read.Bounded` we should only assert that we have seen the READ if we're looking at the TextIO read.

Redone to move the second instanceof into the if, so the add is called regardless of the visit order.